File: kyw0716/2주차/첫 시도/완주하지 못한 선수.js
function solution(participant, completion) {
  let answer = "";
  let hash = {};
  [...new Set([...participant])].forEach((name) => {
    hash[name] = 0;
  });
  participant.forEach((name) => {
    hash[name] += 1;
  });
  completion.forEach((name) => {
    hash[name] -= 1;
  });
  Object.keys(hash).forEach((key) => {
    if (hash[key] === 1) answer = key;
  });

  return answer;
}

File: sk000801/LEVEL1/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        List<String> p = Arrays.asList(participant);
        List<String> c = Arrays.asList(completion);

        for(int i=0; i<participant.length; i++) {
            if(!c.contains(participant[i])) {
                answer = participant[i];
            }
            else if(Collections.frequency(c, participant[i])
                    < Collections.frequency(p,participant[i]))  {
                    answer = participant[i];
                }
            }
        return answer;
    }
}